api/envoy/admin/v2alpha/config_dump.proto

```diff
@@ -0,0 +1,130 @@
+syntax = "proto3";
+
+package envoy.admin.v2alpha;
+
+import "envoy/api/v2/cds.proto";
+import "envoy/api/v2/lds.proto";
+import "envoy/api/v2/rds.proto";
+import "envoy/config/bootstrap/v2/bootstrap.proto";
+
+import "google/protobuf/any.proto";
+
+import "gogoproto/gogo.proto";
+
+// [#protodoc-title: ConfigDump]
+
+// The /config_dump admin endpoint uses this wrapper message to maintain and serve arbitrary
+// configuration information from any component in Envoy.
+// TODO(jsedgwick) In the future, we may want to formalize this further with an RPC for config_dump,
+// and perhaps even with an RPC per config type. That strategy across all endpoints will allow for
+// more flexibility w.r.t. protocol, serialization, parameters, etc.
+message ConfigDump {
+  // This map is serialized and dumped in its entirety at the /config_dump endpoint.
+  //
+  // Keys should be a short descriptor of the config object they map to. For example, Envoy's HTTP
+  // routing subsystem might use "routes" as the key for its config, for which it uses the message
```

Comment:
Did we ever discuss using a type URL here instead? Seems like one less need for magic constants.

Reply:
The type URL doesn't really map exactly to this output, and end-users are not going to be familiar with that detail, so IMO it's best to keep it human readable and disjoint.

Reply:
Sorry I just realize this is an `Any`. It already has the type in it. Is that good enough?

Reply:
Yes, so should it just be a `repeated` instead of `map`?

Reply:
The map gives us the friendly name. And in the future I think we want to be able to do something like:
`config_dump?name=routes`

I suppose we could document all of the full type names, but that seems kind of ugly. TBH I don't feel that strongly about it. If you would rather lose the friendly name for now I can kill it, make it repeated, and we can figure out a friendly name later if needed.